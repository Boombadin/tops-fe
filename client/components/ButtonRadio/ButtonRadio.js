import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const ButtonRadioWrapper = styled.label`
  display: block;
  position: relative;
  padding-left: 25px;
  margin: 0 10px 0 0;
  cursor: pointer;
  font-size: 13px;
  font-weight: 700;
  letter-spacing: -0.4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -webkit-tap-highlight-color: transparent;

  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    z-index: -1;
  }

  &:hover input ~ .radio-checkmark {
    background-color: #eee;
  }

  input:checked ~ .radio-checkmark {
    background-color: white;
    border: 1px solid #cdcdcd;
  }

  input:checked ~ .radio-checkmark:after {
    display: block;
  }
`;

const RadioCheckMark = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 18px;
  width: 18px;
  border-radius: 50%;
  border: 1px solid #ccc;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 50%;
  transform: translateY(-50%);
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -webkit-tap-highlight-color: transparent;

  &:after {
    content: '';
    position: absolute;
    display: none;
  }

  &:after {
    width: 10px;
    height: 10px;
    border-radius: 50%;
    background: #80bd00;
  }
`;

const ButtonRadio = ({
  name,
  label,
  onChange,
  checked = null,
  readOnly = null,
}) => {
  return (
    <ButtonRadioWrapper className="radio-container">
      {label}
      <input
        type="radio"
        name={name}
        checked={checked}
        onChange={onChange}
        readOnly={readOnly}
      />
      <RadioCheckMark className="radio-checkmark"></RadioCheckMark>
    </ButtonRadioWrapper>
  );
};

ButtonRadio.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
};

ButtonRadio.defaultProps = {
  label: '',
};

export default ButtonRadio;
