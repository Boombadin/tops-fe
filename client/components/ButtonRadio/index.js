import ButtonRadio from './ButtonRadio';

export { ButtonRadio }