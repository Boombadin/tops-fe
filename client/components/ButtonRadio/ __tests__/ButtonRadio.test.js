import { render, screen } from '@testing-library/react';
import React from 'react';

import ButtonRadio from '../ButtonRadio';

describe('Component : ButtonRadio', () => {
  const props = {
    name: 'radio',
    label: 'textRadio',
    onChange: jest.fn(),
    checked: null,
    readOnly: null,
  };
  test('Render radio button', () => {
    const { asFragment } = render(<ButtonRadio {...props} />);
    expect(screen.getByText('textRadio')).toBeInTheDocument();
    expect(asFragment(<ButtonRadio {...props} />)).toMatchSnapshot();
  });
  test('Render radio button', () => {
    const props = {
      name: 'radio',
      label: 'textRadio',
      onChange: jest.fn(),
    };
    const { asFragment } = render(<ButtonRadio {...props} />);
    expect(screen.getByText('textRadio')).toBeInTheDocument();
    expect(asFragment(<ButtonRadio {...props} />)).toMatchSnapshot();
  });
  test('checked radio button', () => {
    const input = document.createElement('input');
    input.setAttribute('type', 'radio');
    input.setAttribute('checked', true);
    expect(input).toHaveAttribute('checked', 'true');
    input.setAttribute('checked', false);
    expect(input).toHaveAttribute('checked', 'false');
  });
});
