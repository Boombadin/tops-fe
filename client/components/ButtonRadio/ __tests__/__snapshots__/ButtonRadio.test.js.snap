// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : ButtonRadio Render radio button 1`] = `
<DocumentFragment>
  <label
    class="radio-container ButtonRadio__ButtonRadioWrapper-sc-1c2fqmv-0 jmXno"
  >
    textRadio
    <input
      name="radio"
      type="radio"
    />
    <div
      class="radio-checkmark ButtonRadio__RadioCheckMark-sc-1c2fqmv-1 dRrEo"
    />
  </label>
</DocumentFragment>
`;

exports[`Component : ButtonRadio Render radio button 2`] = `
<DocumentFragment>
  <label
    class="radio-container ButtonRadio__ButtonRadioWrapper-sc-1c2fqmv-0 jmXno"
  >
    textRadio
    <input
      name="radio"
      type="radio"
    />
    <div
      class="radio-checkmark ButtonRadio__RadioCheckMark-sc-1c2fqmv-1 dRrEo"
    />
  </label>
</DocumentFragment>
`;
